### PR TITLE
fixed #1001

### LINKED ISSUE #1001: The first RTP packet that triggers OnTrack wouldn't be received (trivial)
### Your environment.
  - Version: v2.1.18

### What did you do?
Use an audio track (opus), both ends have a local track (bidirectional).
Sending a dummy RTP packet from both ends.

### What did you expect?
The RTP packets should be received by the remote.

### What happened?
The first RTP packet sent by the sender successfully triggered remote's OnTrack() callback, but the packet never gets received.

I know, the RTP stream is not guaranteed delivery but the first packet of sender->receiver always lost. For test purposes, as it is lost always, I have to write the packet more than once - a bit silly.

This is definitely trivial, but it would be nice if the first packet correctly surfaces on track.Read()/ReadRTP().

### PR DESCRIPTION
**Pull request by illuspas**
The determinePayloadType function read the first packet of RTP to get the PayloadType.
Cache this packet and return it on the next read.

### CODE REVIEW

**Comment by Sean-Der** [comment]
Fantastic! This LGTM @illuspas 

Would you mind writing a simple test? Just asserting that if I send two packets both get received?

File: track.go [hunk 1]
```diff
@@ -99,6 +101,12 @@ func (t *Track) Read(b []byte) (n int, err error) {
 
 // ReadRTP is a convenience method that wraps Read and unmarshals for you
 func (t *Track) ReadRTP() (*rtp.Packet, error) {
+	if t.cachePacket != nil {
```
Comment (Sean-Der) [inline track.go:104]: Can you move this into `Read`, then should be good!

Comment (illuspas) [reply track.go:104]: What needs to be cached here is the rtp.Packet structure, so why move to Read?

**Comment by illuspas** [comment]
It does work well in my project.
However I also feel that it is not elegant to fix the problem in this way.
It is better if there are other ways to get the PayloadType in 
```
func (pc *PeerConnection) startReceiver()
```
I haven't found it yet.

File: track.go (hunk 1 above)
Comment (Sean-Der) [reply track.go:104]: This fix only works if the user calls `ReadRTP` if a user calls `Read` they will not get the first packet.

If we move the caching into `Read` then `Read` and `ReadRTP` will be fixed!

Comment (illuspas) [reply track.go:104]: Oh, I see. So it seems that this is not a good way.

**Comment by Sean-Der** [comment]
Hey @illuspas 

I think your solution makes sense. We need to implement this caching somewhere, why not in `pion/webrtc`?

We could add a `Peek` method to `pion/srtp`, but we are just pushing the complexity into another package. IMO I would prefer any custom behavior (like this) to stay in `pion/webrtc`

**Pull request closed without merging**